crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1749,6 +1977,62 @@ impl<T: R2D2Connection> IndexerStore for PgIndexerStore<T> {
             .await
     }
 
+    async fn prune_epoch(&self, epoch: u64) -> Result<(), IndexerError> {
+        let (mut min_cp, max_cp) = match self.get_checkpoint_range_for_epoch(epoch)? {
+            (min_cp, Some(max_cp)) => Ok((min_cp, max_cp)),
+            _ => Err(IndexerError::PostgresReadError(format!(
+                "Failed to get checkpoint range for epoch {}",
+                epoch
+            ))),
+        }?;
+
+        // NOTE: for disaster recovery, otherwise get_transaction_range_for_checkpoint would fail,
+        // as cp_tx table has pruned the cp already.
+        let min_prunable_cp = self.get_min_prunable_checkpoint()?;
+        min_cp = std::cmp::max(min_cp, min_prunable_cp);
+        for cp in min_cp..=max_cp {
+            info!(
+                "Pruning checkpoint {} of epoch {} (min_prunable_cp: {})",
+                cp, epoch, min_prunable_cp
+            );
+            self.execute_in_blocking_worker(move |this| this.prune_checkpoints_table(cp))
```

Comment:
It is crucial that when we prune a cp, checkpoints table is pruned first and `cp_tx` is updated last, and everything else is in between. Maybe add a comment here detailing this importance so we wouldn't easily change this order in the future.